examples/js/loaders/GLTFLoader.js

```diff
@@ -1509,97 +1514,96 @@ THREE.GLTFLoader = ( function () {
 		}
 
 	}
+	function isObjectEqual( a, b ) {
 
-	function isPrimitiveEqual( a, b ) {
-
-		var dracoExtA = a.extensions ? a.extensions[ EXTENSIONS.KHR_DRACO_MESH_COMPRESSION ] : undefined;
-		var dracoExtB = b.extensions ? b.extensions[ EXTENSIONS.KHR_DRACO_MESH_COMPRESSION ] : undefined;
-
-		if ( dracoExtA && dracoExtB ) {
+		if ( Object.keys( a ).length !== Object.keys( b ).length ) return false;
 
-			if ( dracoExtA.bufferView !== dracoExtB.bufferView ) return false;
+		for ( var key in a ) {
 
-			return isObjectEqual( dracoExtA.attributes, dracoExtB.attributes );
+			if ( a[ key ] !== b[ key ] ) return false;
 
 		}
 
-		if ( a.indices !== b.indices ) {
-
-			return false;
+		return true;
 
-		}
+	}
 
-		return isObjectEqual( a.attributes, b.attributes );
+	function createPrimitiveKey( primitiveDef ) {
 
-	}
+		var dracoExtension = primitiveDef.extensions && primitiveDef.extensions[ EXTENSIONS.KHR_DRACO_MESH_COMPRESSION ];
+		var geometryKey;
 
-	function isObjectEqual( a, b ) {
+		if ( dracoExtension ) {
 
-		if ( Object.keys( a ).length !== Object.keys( b ).length ) return false;
+			geometryKey = 'draco:' + dracoExtension.bufferView
+				+ ':' + dracoExtension.indices
+				+ ':' + createAttributesKey( dracoExtension.attributes );
 
-		for ( var key in a ) {
+		} else {
 
-			if ( a[ key ] !== b[ key ] ) return false;
+			geometryKey = primitiveDef.indices + ':' + createAttributesKey( primitiveDef.attributes ) + ':' + primitiveDef.mode;
 
 		}
 
-		return true;
+		return geometryKey;
 
 	}
 
-	function isArrayEqual( a, b ) {
+	function createAttributesKey( attributes ) {
 
-		if ( a.length !== b.length ) return false;
+		var attributesKey = '';
 
-		for ( var i = 0, il = a.length; i < il; i ++ ) {
+		var keys = Object.keys( attributes ).sort();
 
-			if ( a[ i ] !== b[ i ] ) return false;
+		for ( var i = 0, il = keys.length; i < il; i ++ ) {
+
+			attributesKey += keys[ i ] + ':' + attributes[ keys[ i ] ] + ';';
 
 		}
 
-		return true;
+		return attributesKey;
 
 	}
 
-	function getCachedGeometry( cache, newPrimitive ) {
+	function createArrayKeyBufferGeometry( a ) {
 
-		for ( var i = 0, il = cache.length; i < il; i ++ ) {
+		var arrayKey = '';
 
-			var cached = cache[ i ];
+		for ( var i = 0, il = a.length; i < il; i ++ ) {
 
-			if ( isPrimitiveEqual( cached.primitive, newPrimitive ) ) return cached.promise;
+			arrayKey += i + a[ i ].uuid;
```

Comment:
minor: why use `i` in the key?

Reply:
Just as a separator, do you think I can drop it completely?

Reply:
As a separator I'd use something that can't occur in the UUID itself, like a ':'.